lib/puppet-strings/markdown/table_of_contents.rb

```diff
@@ -3,24 +3,17 @@
 module PuppetStrings::Markdown
   module TableOfContents
     def self.render
-      final = "## Table of Contents\n\n"
+      template = File.join(File.dirname(__FILE__), 'templates/table_of_contents.erb')
+      renders = PuppetStrings::Markdown.groups.map do |group_info|
+        group_name = group_info.name
+        group = group_info.items.map { |item| item.toc_info }
+        has_private = group.any? { |item| item[:private] }
+        has_public = group.any? { |item| !item[:private] }
 
-      [PuppetStrings::Markdown::PuppetClasses,
-      PuppetStrings::Markdown::DefinedTypes,
-      PuppetStrings::Markdown::ResourceTypes,
-      PuppetStrings::Markdown::Functions,
-      PuppetStrings::Markdown::DataTypes,
-      PuppetStrings::Markdown::PuppetTasks,
-      PuppetStrings::Markdown::PuppetPlans].each do |r|
-        toc = r.toc_info
-        group_name = toc.shift
-        group = toc
-        priv = r.contains_private?
-
-        template = File.join(File.dirname(__FILE__), 'templates/table_of_contents.erb')
-        final += PuppetStrings::Markdown.erb(template).result(binding)
+        PuppetStrings::Markdown.erb(template).result(binding)
```

Comment:
I wonder if you can also store this outside of the loop (just like the `template` variable) and render it multiple times

Reply:
Good question!